src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+        self._MirrorDescanner = MirrorDescanner("MirrorDescanner", role=None, parent=self)
+        self.children.value.add(self._MirrorDescanner)
+
+        try:
+            kwargs = children["EBeamScanner"]
+        except Exception:
+            raise ValueError("Required child EBeamScanner not provided")
+        self._EBeamScanner = EBeamScanner("EBeamScanner", role=None, parent=self)
+        self.children.value.add(self._EBeamScanner)
+
+        try:
+            kwargs = children["MPPC"]
+        except Exception:
+            raise ValueError("Required child MPPC not provided")
+        self._MPPC = MPPC("MPPC", None, self)
+        self.children.value.add(self._MPPC)
+
+    @classmethod
+    def simpleASMCall(cls, url, method, expected_status, data=None, raw_response=False, timeout=600):
+        """
+
+        :param url: url of the command, server part is defined in global variable URL
+        :param method: getting or posting via global variables _METHOD_GET/_METHOD_POST
+        :param expected_status: expected feedback of server for a positive call
+        :param data: data added to the call
+        :param raw_response: specified the format of the structure returned
+        :param timeout: [s] if within this period no bytes are received an timeout exception is raised
+        :return: status_code(posting), content dictionary(getting), or entire response (raw_response=True)
+        """
+        logging.getLogger().setLevel(logging.INFO)
+        logging.info("Executing: %s" % url)
+        if method == _METHOD_GET:
+            resp = requests.get(url, json=data, timeout=timeout)
+        elif method == _METHOD_POST:
+            resp = requests.post(url, json=data, timeout=timeout)
+
+        if resp.status_code != expected_status:
+            raise AsmApiException(url, resp, expected_status)
+
+        logging.info("Call to %s went fine, no problems occured\n" % url)
+
+        if raw_response:
+            return resp
+        elif method == _METHOD_POST:
+            return resp.status_code
+        elif method == _METHOD_GET:
+            return json.loads(resp.content)
+
+
+class EBeamScanner(model.Emitter):
+
+    def __init__(self, name, role, parent, **kwargs):
+        super(EBeamScanner, self).__init__(name, role, **kwargs)
+        self.parent = parent
+
+        # self._metadata[model.MD_PIXEL_SIZE] =
+
+        clockFrequency = self.parent.simpleASMCall(self.parent.server_URL + "/scan/clock_frequency", _METHOD_GET, 200)[
+            'frequency']
+        self.clockFrequency = model.IntVA(clockFrequency, unit='Hz')
+        self.clockPeriod = model.FloatVA(1 / self.clockFrequency.value, unit='s')
+        self._shape = model.TupleVA((6400, 6400), unit='px,px')
+        self.resolution = model.ResolutionVA((6400, 6400), ((10, 10), (7200, 7200)), setter=self._setResolution)
+        self.dwellTime = model.FloatContinuous(float(self.clockPeriod.value), (float(self.clockPeriod.value), 100000.0),
```

Comment:
That is actually usually equal to the clockperiod, but just in case I will also add the 400 ns.
We need to have the clockperiod as a minimum because the dwellTime can otherwise become zero because of rounding (dwelltime is rounded to a multiple of the clockPeriod, because the "unit" is clockperiods)

Reply:
I added these time restrictions, and also added that the minimum should be at last one clockperiod (that is the unit in which we give it to the API)

Reply:
I wouldn't use one clock period as the minimum for the VA (25ns). This would mean the user can actually put that number in the GUI as dwell time. Job mentioned that below 400ns will not be possible in EA1. So we also don't want the user to select this. So I would suggest to use Job's recommendation of a range of (400ns to 100us).
min = clockPeriod.value *16